pymeasure/adapters/ni_gpib_232ct.py

```diff
@@ -0,0 +1,345 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+from enum import IntFlag, Flag
+import time
+from pymeasure.adapters import VISAAdapter
+from pymeasure.instruments.validators import strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+# pylint: disable=W1203
+
+
+class NI_GPIB_232(VISAAdapter):
+    """Encapsulates the additional commands necessary
+       to communicate over a National Instruments GPIB-232CT Adapter,
+       using the :class:`VISAAdapter`.
+
+       Each Adapter is constructed based on a connection to the device
+       itself and the GPIB address of the instrument to be communicated with.
+       Connection sharing is achieved by using the :meth:`.gpib`
+       method to spawn new NI_GPIB_232s for different GPIB addresses.
+
+       :param resource_name: A VISA resource string that identifies the
+           connection to the device itself, for example "ASRL5" for the 5th COM port.
+       :param address: Integer GPIB address of the desired instrument.
+       :param serial_timeout: Timeout value for the serial communication
+       :param rw_delay: delay value defining the internal waiting time
+       :param eoi: Enable or disable EOI assertion.
+       :param kwargs: Key-word arguments if constructing a new serial object
+       :ivar address: Integer GPIB address of the desired instrument.
+
+
+       Usage example:
+
+       .. code::
+
+         adapter = NI_GPIB_232("ASRL5::INSTR", 7)
+         sourcemeter = Keithley2400(adapter)  # at GPIB address 7
+         # generate another instance with a different GPIB address:
+         adapter2 = adapter.gpib(9)
+         multimeter = Keithley2000(adapter2)  # at GPIB address 9
+
+
+    """
+
+    def __init__(self, resource_name, address=None, serial_timeout=1000, rw_delay=50,
```

Comment:
We intend to harmonize at least timeouts toward milliseconds (#783), so that is the correct choice. Units should be documented in the docstring, though, for additional clarity.

Reply:
docstrings will be updated in a commit soon